src/components/ebay-filter-menu/index.js

```diff
@@ -0,0 +1,80 @@
+const assign = require('core-js-pure/features/object/assign');
+const scrollKeyPreventer = require('makeup-prevent-scroll-keys');
+const rovingTabindex = require('makeup-roving-tabindex');
+const eventUtils = require('../../common/event-utils');
+const template = require('./template.marko');
+
+module.exports = require('marko-widgets').defineComponent({
+    template,
+    getInitialState(input) {
+        return assign({}, input, {
+            items: (input.items || []).map(item => assign({}, item))
+        });
+    },
+    onRender(event) {
+        this.contentEl = this.el.querySelector('.filter-menu__menu, .filter-menu-button__menu');
+
+        if (this.state.withButton) {
+            this.contentEl = this.el;
+        }
+
+        if (event.firstRender) {
+            if (this.state.variant !== 'form') {
+                // FIXME: should be outside of firstRender, but only when itemEls changes
+                this.rovingTabindex = rovingTabindex.createLinear(
+                    this.contentEl.querySelector('[role="menu"]'), 'div',
+                    { index: 0, autoReset: 0 }
+                );
+
+                scrollKeyPreventer.add(this.contentEl);
+            }
+        }
+    },
+    onDestroy() {
+        scrollKeyPreventer.remove(this.contentEl);
+    },
+    setCheckedItem(itemIndex, itemEl) {
+        const item = this.state.items[itemIndex];
+
+        if (item) {
+            this.state.items[itemIndex].checked = !item.checked;
+            this.setStateDirty('items');
+            this.emitComponentEvent('change', itemEl);
+        }
+    },
+    getCheckedItems() {
+        return this.state.items
+            .filter(item => item.checked)
+            .map(item => item.value);
+    },
+    handleItemClick(e, itemEl) {
+        this.setCheckedItem(this.getItemElementIndex(itemEl), itemEl);
+    },
+    handleItemKeydown(e, itemEl) {
+        eventUtils.handleEscapeKeydown(e, () => {
+            this.emitComponentEvent('keydown', null, e);
+        });
+
+        if (this.state.variant !== 'form') {
+            eventUtils.handleActionKeydown(e, () => {
+                this.setCheckedItem(this.getItemElementIndex(itemEl), itemEl);
+            });
+        }
+    },
+    handleFooterButtonClick(originalEvent) {
+        this.emitComponentEvent('footer-click', null, originalEvent);
+    },
+    handleFormSubmit(originalEvent) {
```

Comment:
Yes, they bubble. `filter-menu` must be able to stand alone, and it's events all likely bubble through the `filter-menu-button`. Developers will likely not be listening to the `filter-menu` events because they will have only implemented the `filter-menu-button`, and be listening for its events.

Let me know if you still have any confusion. I'd be happy to talk through things.